### ISSUE TITLE
Starbound Player List Doesn't display

### ISSUE THREAD

**Issue by Ruok2bu** [Austinb/GameQ#186]
The player list of a starbound server isn't displaying.

Here's what im testing against:

array(
        'id' => 'Starbound',
        'type' => 'starbound',
        'host' => '192.99.15.93:21025',
    ),

// Some functions to print the results
function print_results($results) {

```
foreach ($results as $id => $data) {

printf("<h2>%s</h2>\n", $id);
echo '<pre>'; print_r($data['players']);
print_table($data);
```

}

**Comment by globeFrEak** [comment]
the player list is empty!
example here: http://dev.cwclan.de/infusions/gameserver_query_panel/gameserver_query_detail.php?id=68

**Comment by Ruok2bu** [comment]
It's definitely possible to get it too because GameTracker is able to get it somehow...

http://www.gametracker.com/server_info/s1.playstarboundrp.com:21025/

**Comment by Austinb** [comment]
I have a note about this on the wiki.  It seems starbound does not follow the standard Source query protocol (uses old or custom) because raw responses do not provide any player or rule information.   If anyone has any more information on the protocol and how it differs from the Valve documentation please link it.   I do not have any spare time to hunt down documentation.

**Comment by Ruok2bu** [comment]
Hopefully this will help you; I ran Wireshark while GameTracker was scanning my server.  here's the output (First log is GameTracker sending the query and the second is my server sending the info):

https://dl.dropboxusercontent.com/u/18722727/Wireshark-Log.txt

**Comment by Austinb** [comment]
Just to update I figured out that starbound is using the old method of getting a challenge - https://developer.valvesoftware.com/wiki/Server_queries#A2S_SERVERQUERY_GETCHALLENGE.  When I changed that I am getting a challenge response now but I am still not getting a response for players or rules from the server when using that challenge string.  So I am not sure where it is failing.  I dont have any more time now to spend on it unfortunately.
